ukis_pysat/raster.py

```diff
@@ -134,31 +127,38 @@ def _pad_to_bbox(self, bbox, mode="constant", constant_values=0):
         max_diff_ll = np.max(np.subtract(tuple(self.dataset.bounds[:2]), bbox[:2]))
         max_diff = max(max_diff_ll, max_diff_ur)  # buffer in units
 
-        pad_width = math.ceil(max_diff / self.transform.to_gdal()[1])  # units / pixel_size
+        pad_width = math.ceil(max_diff / self.dataset.transform.to_gdal()[1])  # units / pixel_size
 
         destination = np.zeros(
             (self.dataset.count, self.__arr.shape[1] + 2 * pad_width, self.__arr.shape[2] + 2 * pad_width,),
             self.__arr.dtype,
         )
 
         for i in range(0, self.dataset.count):
-            destination[i], self.transform = rasterio.pad(
-                self.__arr[0], self.transform, pad_width, mode, constant_values=constant_values,
+            destination[i], transform = rasterio.pad(
+                self.__arr[0], self.dataset.transform, pad_width, mode, constant_values=constant_values,
             )
 
         self.__arr = destination
 
-        return self.__update_dataset(self.dataset.meta)
+        return self.__update_dataset(self.dataset.meta, self.dataset.crs, transform)
 
-    def __update_dataset(self, meta):
+    def __update_dataset(self, meta, crs, transform):
         """Update dataset without writing to file after it theoretically changed.
 
         :param meta: The basic metadata of the dataset as returned from the meta property of rasterio Datasets
         :return: file in memory
         """
 
         meta.update(
-            {"height": self.__arr.shape[-2], "width": self.__arr.shape[-1], "transform": self.transform,}
+            {
+                "height": self.__arr.shape[-2],
+                "width": self.__arr.shape[-1],

```

Comment:
is it not saver to put also "count": self.__arr.shape[0] here to make sure to read the correct band count from array. this may differ from dataset.count if band count has been updated by stacking arrays together along the z-axis.

Reply:
Good point, I will put everything in there.